rtc_data_service/src/auth_enclave_actor.rs

```diff
@@ -0,0 +1,67 @@
+use actix::prelude::*;
+use rtc_uenclave::{AttestationError, EnclaveConfig, RtcAuthEnclave};
+use std::sync::Arc;
+
+#[derive(Default)]
+pub(crate) struct RequestAttestation;
+
+type RequestAttestationResult = Result<String, AttestationError>;
+
+impl Message for RequestAttestation {
+    type Result = RequestAttestationResult;
+}
+
+pub struct AuthEnclaveActor {
+    enclave: Option<RtcAuthEnclave<Arc<EnclaveConfig>>>,
+    config: Arc<EnclaveConfig>,
+}
+
+impl AuthEnclaveActor {
+    pub fn new(config: Arc<EnclaveConfig>) -> Self {
+        Self {
+            enclave: None,
+            config,
+        }
+    }
+
+    pub(crate) fn get_enclave(&self) -> &RtcAuthEnclave<Arc<EnclaveConfig>> {
+        self.enclave
+            .as_ref()
+            .expect("AuthEnclaveActor: tried to access enclave while not initialised")
+    }
+}
+
+impl Drop for AuthEnclaveActor {
```

Comment:
This is some debug things that went in when I were looking at a double free on SIGTERM handling. We can remove this for the auth enclave and data enclave actors.

Reply:
(a2944f5 drops this for the auth enclave, but the data enclave's one still needs removing)